data_pipeline/twitter/data_from_twitter_rest_api.py

```diff
@@ -0,0 +1,302 @@
+import time
+import logging
+from typing import Iterable
+
+import tweepy
+from google.cloud.bigquery.client import Client
+from tweepy import API, User
+from tweepy.cursor import BaseIterator
+
+from data_pipeline.twitter.process_tweet_response import (
+    extract_tweet_from_twitter_response,
+    modify_by_search_term_occurrence_location_in_response
+)
+from data_pipeline.twitter.twitter_bq_util import (
+    batch_load_iter_record_to_bq, batch_load_multi_iter_record_to_bq
+)
+from data_pipeline.twitter.twitter_config import (
+    TwitterDataPipelineConfig, TweetType,
+    TwitterPipelineModuleConstants, ETL_TIMESTAMP_FORMAT
+)
+from data_pipeline.utils.data_store.bq_data_service import get_bq_table
+from data_pipeline.utils.pipeline_file_io import WriteIterRecordsInBQConfig
+
+from data_pipeline.utils.record_processing import (
+    add_provenance_to_record, standardize_record_keys
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _limit_handled(cursor: BaseIterator):
+    while True:
+        try:
+            yield cursor.next()
+        except tweepy.RateLimitError:
+            time.sleep(15 * 60)
+        except StopIteration:
+            break
+
+
+class TwitterRestApi:
+    def __init__(
+            self, auth_oauth
+    ):
+        self.tweepy_api = API(auth_oauth)
+
+    def get_api_restricted_followers_of_user(self, twitter_username: str):
+        twitter_username = twitter_username.rstrip('@')
+        for followers in _limit_handled(
+                tweepy.Cursor(
+                    self.tweepy_api.followers, id=twitter_username, count=1000
+                ).pages()
+        ):
+            for follower in followers:
+                yield extract_user_properties_to_dict(follower)
+
+    def get_latest_api_restricted_tweets_from_user(self, twitter_username):
+        twitter_username = twitter_username.rstrip('@')
+        for tweets in _limit_handled(
+                tweepy.Cursor(
+                    self.tweepy_api.user_timeline, id=twitter_username
+                ).pages()
+        ):
+            for tweet in tweets:
+                yield tweet
+
+    def get_all_retweets_of_a_tweet(self, tweet_id: int):
+        try:
+            retweets_list = self.tweepy_api.retweets(
+                tweet_id, tweet_mode='extended', count=10000
+            )
+        except tweepy.RateLimitError:
+            time.sleep(15 * 60)
+            retweets_list = self.tweepy_api.retweets(
+                tweet_id, tweet_mode='extended', count=10000
+            )
+
+        for retweet in retweets_list:
```

Comment:
This sounds like the question you had the other day. You could write it as `yield from retweets_list`. Although the slightly more verbose way may be more obvious when not being familiar with yield from.
(In this particular case we probably could even return `retweets_list` itself)